aptos-move/aptos-aggregator/src/aggregator_extension.rs

```diff
@@ -189,14 +246,15 @@ impl Aggregator {
                 if self.value <= value {
                     self.value = subtraction(value, self.value)?;
                     self.state = AggregatorState::PositiveDelta;
+                    if self.value > self.limit {
```

Comment:
I think this is a bug? tell me if I'm wrong:
it seems like there was a silent assumption before that the value won't break the limit (@georgemitenkov ?), but we should never allow the value to be set above the limit, right? Right now, we would record overflow, but actually allow overflow to happen.

If the above is right, then we would have to perform subtraction, check the result vs the limit, if it's below the limit set it, otherwise, don't set it, record overflow, and return an error from try_add?

Reply:
You are right. I updated the self.value and then raised an overflow. I should have raised an overflow before updating self.value. Fixing it.